pywbem/cim_operations.py

```diff
@@ -1337,112 +1326,135 @@ def _configure_detail_level(cls, detail_level):
-                                                     LOG_DETAIL_LEVELS))
+                raise ValueError("Invalid log detail level string: %r; must "
+                                 "be one of: %s" %
+                                 (detail_level, LOG_DETAIL_LEVELS))
         elif isinstance(detail_level, int):
             if detail_level < 0:
-                raise ValueError('Log detail_level integer input must be '
-                                 'positive integer. "%s" rejected' %
-                                 detail_level)
+                raise ValueError("Invalid log detail level integer: %r; must "
+                                 "be a positive integer." % detail_level)
 
         elif detail_level is None:
-            # pylint: disable=redefined-variable-type
             detail_level = DEFAULT_LOG_DETAIL_LEVEL
         else:
-            raise ValueError('log http detail level "%s" invalid. Must be in '
-                             ' %sor positive integer'
-                             % (detail_level, LOG_DETAIL_LEVELS))
+            raise ValueError("Invalid log detail level: %r; must be one of: "
+                             "%s, or a positive integer" %
+                             (detail_level, LOG_DETAIL_LEVELS))
         return detail_level
 
     @classmethod
     def _configure_logger_handler(cls, log_dest, log_filename):
         """
-        Configure the log destination and return the defined handler.
+        Return a logging handler for the specified log_dest,
+        or None if log_dest is None.
         """
-        msg_format = '%(asctime)s-%(name)s-%(message)s'
-        if not log_dest:
+
+        if log_dest is None:
             return None
-        if log_dest not in LOG_DESTINATIONS:
-            raise ValueError('Invalid log destination "%s". valid log '
-                             'destinations are: %s' %
-                             (log_dest, LOG_DESTINATIONS))
 
-        # process dest parameter
+        msg_format = '%(asctime)s-%(name)s-%(message)s'
+
         if log_dest == 'stderr':
-            # Cannot use the stream parameter because not defined for
-            # python 2.6. Removed stream=sys.stderr
+            # TODO: stream=sys.stderr parameter is not defined in py26
             handler = logging.StreamHandler()
             handler.setFormatter(logging.Formatter(msg_format))
         elif log_dest == 'file':
             if not log_filename:
-                raise ValueError('Filename required if log destination '
-                                 'is "file"')
-            # pylint: disable=redefined-variable-type
+                raise ValueError("Log filename is required if log destination "
+                                 "is 'file'")
             handler = logging.FileHandler(log_filename, encoding="UTF-8")
             handler.setFormatter(logging.Formatter(msg_format))
-        elif log_dest == 'none' or log_dest is None:
+        elif log_dest == 'none':
```

Comment:
We should actually have a default of having a null handler, and don't need a means to explicitly set up the null handler.

Reply:
DONE - Removed the support for 'none' and added an according TODO for the best practices.